src/jetstream/auth.go

```diff
@@ -102,11 +102,32 @@ func (p *portalProxy) initSSOlogin(c echo.Context) error {
 		return err
 	}
 
+	if !safeState(state, p.Config.SSOWhiteList) {
+		err := interfaces.NewHTTPShadowError(
+			http.StatusUnauthorized,
+			"SSO Login: Disallowed state",
+			"SSO Login: Disallowed state")
+		return err
+	}
+
 	redirectURL := fmt.Sprintf("%s/oauth/authorize?response_type=code&client_id=%s&redirect_uri=%s", p.Config.ConsoleConfig.AuthorizationEndpoint, p.Config.ConsoleConfig.ConsoleClient, url.QueryEscape(getSSORedirectURI(state, state, "")))
 	c.Redirect(http.StatusTemporaryRedirect, redirectURL)
 	return nil
 }
 
+func safeState(state string, whiteListStr string) bool {
+	if len(whiteListStr) == 0 {
+		return true;
+	}
+
+	whiteList := strings.Split(whiteListStr, ",")
+	if len(whiteList) == 0 {
+		return true;
+	}
+
+	return ArrayContainsString(whiteList, state)
```

Comment:
I don't think you can do an exact match check - either we use regex or we accept a URL that at least starts with any of the whitelist URLs